page.js

```diff
@@ -49,256 +51,281 @@ const token_id = "personal-dj-token";
+    "click",
+    function (e) {
+      e.preventDefault();
+
+      let seed_artists = trackResult[0].artists[0].id;
+      let seed_tracks = selectedTrackId;
+
+      // Check selected track and artist exists
+      if (
+        !seed_tracks ||
+        seed_tracks == "" ||
+        !seed_artists ||
+        seed_artists == ""
+      ) {
+        alert("Please select a track first.");
+      }
+
+      let dance = "danceability";
+      let energy = "energy";
+      let popular = "popular";
+      let limit = "limit";
+
+      // Validate the dance, energy, popular, and limit
+      // Using booleans like this because we can validate multiple inputs at a time instead of 1 input at a time
+      danceValid = validateForm(dance);
+      energyValid = validateForm(energy);
+      popularValid = validateForm(popular);
+      limitValid = validateForm(limit);
+
+      if (!danceValid || !energyValid || !popularValid || !limitValid) {
+        return;
+      }
+
+      dance = document.getElementById("danceability").value;
+      energy = document.getElementById("energy").value;
+      popular = parseInt(document.getElementById("popular").value);
+      limit = parseInt(document.getElementById("limit").value);
+
+      // Check for invalid inputs
+      if (
+        !limit ||
+        !Number.isInteger(limit) ||
+        limit < 1 ||
+        limit > 50 ||
+        !seed_artists ||
+        seed_artists.length < 1 ||
+        !seed_tracks ||
+        seed_tracks.length < 1 ||
+        !dance ||
+        dance < 0 ||
+        dance > 10 ||
+        !energy ||
+        energy < 0 ||
+        energy > 10 ||
+        !popular ||
+        !Number.isInteger(popular) ||
+        popular < 1 ||
+        popular > 100
+      ) {
+        return;
+      }
+
+      // change energy and hype to decimal values
+      dance = parseFloat(dance / 10);
+      energy = parseFloat(energy / 10);
+
+      const { originalText: origText, onRemoveSpinner } = loading(
+        "rec-button",
+        true
+      );
 
-    // search for a track by name
-    function searchTrackByName(track_name, offset = 0, onDone) {
       $.ajax({
-        url: "/trackSearch",
+        url: "/recommendations",
         data: {
-          track_value: track_name,
-          searchOffset: offset,
+          limit: limit,
+          seed_artists: seed_artists,
+          seed_tracks: seed_tracks,
+          danceability: dance,
+          energy: energy,
+          popular: popular,
           token: sessionStorage.getItem(token_id),
         },
       }).done(function (data) {
         if (responseIsSuccess(data)) {
-          trackResult = data.trackResult.tracks.items;
-          displaySearchResults(trackResult);
-        }
-        if (onDone && typeof onDone === "function") {
-          onDone(data);
+          recList_id = displayRecommendations(data.trackResult);
+          recList_cache = data.trackResult;
+
+          recList_id = displayRecommendations(data.trackResult);
```

Comment:
Line #239 and #242 seem like they do the exact same thing, did you mean to set the filter_flag for one of them?

Reply:
No, It must be something leftover from the merge conflict. I will just remove it, seems like a silly mistake.